### DISCUSSION TITLE
Can we initializing / modify context directly?

### DISCUSSION THREAD

**Discussion by github0013** [statelyai/xstate#2472] [Q&A] [answered]
https://xstate.js.org/docs/guides/context.html#initial-context

This page shows context can be updated by

- actions
- machine factory functions
- withContext

However, I can still directly edit the context values like this.

```tsx
import { useMachine } from "@xstate/react"
import { createMachine } from "xstate"

const machine = createMachine<{value: number}>({
  context: {
    value: 0
  }
})

...

  const [state, send] = useMachine(machine)

  React.useEffect(() => {
    state.context.value = 42
  }, [])

```

Is this allowed to perform?

**Comment by Andarist** [comment, marked as answer]
> Is this allowed to perform?

No, this is not recommended. It might work because we don't freeze the context object but all modifications to context should be done from within the machine and not using an external mutation.

**Comment by hiroMTB** [comment]
I did it like this way

```
const value = 42

// we make a new machine with new context value
const newMachine = machine.withContext({ value })

// This is current state
const currentStateValue = service.state.value

// make new state with new context
const state = newMachine.getInitialState(currentStateValue)

// restart service
service.stop()
service.start(state)
```
